src/ProxyFactory.php

```diff
@@ -58,19 +58,32 @@ public function __construct(
     /**
      * @template T of object
      *
-     * @param T $object
+     * @param class-string<T> $class
+     * @param mixed ...$args
      *
      * @return T
      */
-    public function createProxy(object $object): object
+    public function createInstance(string $class, ...$args): object
     {
-        $instanceRef = new \ReflectionObject($object);
-        $methods = $instanceRef->getMethods(\ReflectionMethod::IS_PUBLIC);
+        $instanceRef = new \ReflectionClass($class);
```

Comment:
although scanning all methods of a class permits us to throw errors on decorated private methods, the cost of loading annotation for each method maybe high in dev env